floss/main.py

```diff
@@ -1027,7 +1021,18 @@ def main(argv=None):
     if not is_workspace_file(sample_file_path):
         if not options.no_static_strings and not options.functions:
             floss_logger.info("Extracting static strings...")
-            print_static_strings(sample_file_path, min_length=min_length, quiet=options.quiet)
+            if os.path.getsize(sample_file_path) > sys.maxsize:
+                floss_logger.warning("File too large, strings listings may be truncated.")
+                floss_logger.warning("FLOSS cannot handle files larger than 4GB on 32bit systems.")
+
+            file_buf = get_file_as_mmap(sample_file_path)
+            print_static_strings(file_buf, min_length=min_length, quiet=options.quiet)
+            static_ascii_strings = strings.extract_ascii_strings(file_buf, min_length)
```

Comment:
can you extract these once and then use for printing and JSON exporting?

Reply:
My thought was to try to keep the memory footprint as small as possible with static strings. There can be a lot of them I chose to use a generator for printing and then create a new one for use with `chain` for JSON serialization. The use of a generator is also what brought me to choose `simplejson` over `json` from the standard lib.

I considered using `itertools.tee` to "fork" the generator but, after reading the documentation, if you let one of the "forks" get too far ahead of the other, you don't end up saving any memory and it becomes more efficient to just use lists or create two generators. 

Ultimately, I'd be looking to you to make the final call on this because I'm torn.

Reply:
Gotcha, thanks! Your explanation makes sense.